metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py

```diff
@@ -104,6 +115,39 @@ def authenticator_type_is_valid(cls, v, values, **kwargs):
                         f"'private_key_path' was none "
                         f"but should be set when using {v} authentication"
                     )
+            elif v == "OAUTH_AUTHENTICATOR":
+                if values.get("oauth_client_id") is None:
+                    raise ValueError(
+                        f"'oauth_client_id' is none "
+                        f"but should be set when using {v} authentication"
+                    )
+                if values.get("oauth_scope") is None:
+                    raise ValueError(
+                        f"'oauth_scope' was none "
+                        f"but should be set when using {v} authentication"
+                    )
+                if values.get("oauth_authority_url") is None:
+                    raise ValueError(
+                        f"'oauth_authority_url' was none "
+                        f"but should be set when using {v} authentication"
+                    )
+                if values.get("use_certificate") is True:
+                    if values.get("base64_encoded_oauth_private_key") is None:
+                        raise ValueError(
+                            f"'base64_encoded_oauth_private_key' was none "
+                            f"but should be set when using {v} authentication"
```

Comment:
This message needs to be changed and add it is needed when `use_certificate` is true